### ISSUE TITLE
Plot recall presicion curve

### ISSUE THREAD

**Issue by FrancoBautista93** [vivinastase/voxseg#1]
I want to plot the recall presicion curve using the 'get_threshold' function in utils but I don´t understand what is the format and how obtain 'y_pred' and 'y_true' arguments. Can you give me a hand with that?

Also, do you know how plot the ROC curve? just changing presicion_recall_curve for roc_curve?

**Comment by vivinastase** [comment]
Hi Franco

get_threshold returns the threshold that gives the best F1 score in training. It is computed based on the predictions on the development data, or if there was no dev part, on the training. The predictions -- y_pred -- are probabilities (for each label), and y_true are the labels. This function is called in the train.py function. Here is how it looks like when called on the full training data:

utils.get_threshold(voxseg_model.model.predict(X[:,:,:,:,np.newaxis]), y)

y are the labels read from the file, and X is the training data (after all the transformations and normalization and all that). You can see how y and X are obtained from the Kladi style input files in the train.py function.

You can't use the get_threshold to plot the precision-recall curve, because it returns only the threshold for which the F1 is the highest. But if you look into the get_threshold function, you can see how to get precision and recall values for a range of thresholds -- using the precision_recall_curve function in the sklearn.metrics package. The ROC curve plots the true positive rate vs. the false positive rate. I think sklearn has some functions to obtain those numbers as well. I hope this helps!

**Comment by FrancoBautista93** [comment]
It does! thank you very much for your help.

**Comment by vivinastase** [comment]
The get_threshold method is passed two arguments, predicted labels and true labels in this order. Then inside this function, the precision_recall_curve method is called on the true labels and predicted labels, in this order. The two functions don't have to have the same order in their arguments, as long as they are used properly :)
